test/test_Factors.cpp

```diff
@@ -75,3 +89,23 @@ TEST_F(FactorsTest, it_returns_position_1000_at_SI_position_max) {
 TEST_F(FactorsTest, it_clamps_the_relative_position_at_plus_1000) {
     ASSERT_FLOAT_EQ(1000, factors.relativePositionFromSI(85));
 }
+
+TEST_F(FactorsTest, it_returns_the_SI_position_for_no_joint_state_position_source) {
+    ASSERT_FLOAT_EQ(84,
+        factors.positionToSI(1000,
+            JointStatePositionSources::JOINT_STATE_POSITION_SOURCE_NONE));
```

Comment:
If there is no position source ... we should definitely not return a position ....